apm-agent-core/src/main/java/co/elastic/apm/agent/impl/payload/SystemInfo.java

```diff
@@ -191,6 +191,7 @@ SystemInfo parseContainerId(String line) {
                         for (int i = 1; i <= matcher.groupCount(); i++) {
                             String podUid = matcher.group(i);
                             if (podUid != null && !podUid.isEmpty()) {
+                                podUid = podUid.replace('_', '-');
```

Comment:
The hyphens are only escaped when using the systemd cgroup driver:

https://github.com/kubernetes/kubernetes/blob/6d0994fa66e34db874e4e26c86b55ba34e75d1c1/pkg/kubelet/cm/cgroup_manager_linux.go#L86

I think we should make this conditional on the path ending with ".scope"

Reply:
Sure thing. See my latest commit.